packages/ses/src/make-evaluate.js

```diff
@@ -31,80 +35,38 @@ function buildOptimizer(constants, name) {
-  //       evaluated code to lookup free variable names on the other scope
-  //       objects and not in global scope.
-  //    b) `moduleLexicals` which provide a way to introduce free variables
-  //       that are not available on the globalObject.
-  //    c) `globalObject` is the global scope object of the evaluator, aka the
-  //       Compartment's `globalThis`.
-  //    d) `scopeTerminator` is a proxy object which prevents free variable
-  //       lookups to escape to the start compartment's global object.
-  // 2: `optimizer`s catch constant variable names for speed.
-  // 3: The inner strict `evaluate` function should be passed two parameters:
-  //    a) its arguments[0] is the source to be directly evaluated.
-  //    b) its 'this' is the this binding seen by the code being
-  //       directly evaluated (the globalObject).
-
-  // Notes:
-  // - The `optimizer` strings only lookup values on the `globalObject` and
-  //   `moduleLexicals` objects by construct. Keywords like 'function' are
-  //   reserved and cannot be used as a variable, so they are excluded from the
-  //   optimizer. Furthermore to prevent shadowing 'eval', while a valid
-  //   identifier, that name is also explicitly excluded.
-  // - when 'eval' is looked up in the `evalScope`, the powerful unsafe eval
-  //   intrinsic is returned after automatically removing it from the
-  //   `evalScope`. Any further reference to 'eval' in the evaluate string will
-  //   get the tamed evaluator from the `globalObject`, if any.
+  return function evaluate(code) {
+    'use strict';
 
-  // TODO https://github.com/endojs/endo/issues/816
-  // The optimizer currently runs under sloppy mode, and although we doubt that
-  // there is any vulnerability introduced just by running the optimizer
-  // sloppy, we are much more confident in the semantics of strict mode.
-  // The `evaluate` function can be and is reused across multiple evaluations.
-  // Since the optimizer should not be re-evaluated every time, it cannot be
-  // inside the `evaluate` closure. While we could potentially nest an
-  // intermediate layer of `() => {'use strict'; ${optimizers}; ...`, it
-  // doesn't seem worth the overhead and complexity.
-  const evaluateFactory = FERAL_FUNCTION(`
-    with (this.scopeTerminator) {
-      with (this.globalObject) {
-        with (this.moduleLexicals) {
-          with (this.evalScope) {
-            ${globalObjectOptimizer}
-            ${moduleLexicalOptimizer}
-            return function() {
-              'use strict';
-              return eval(arguments[0]);
-            };
-          }
-        }
-      }
-    }
-  `);
+    context.evalScope.eval; // required by make-safe-evaluator.
 
-  return apply(evaluateFactory, context, []);
+    return apply(
+      FERAL_FUNCTION(
+        `{${SCOPE_TERMINATOR_KEYS}}`,
+        `{${MODULE_LEXICAL_KEYS}}`,
+        `return (function () {
+          'use strict';
+          ${code}
+        }).bind(this)();`,
```

Comment:
simply prepending `'use strict';` to `code` sadly doesn't work because of the destructuring of arguments

`SyntaxError: Illegal 'use strict' directive in function with non-simple parameter list `